cli/proxy.go

```diff
@@ -106,6 +98,15 @@ var proxyCommand = cli.Command{
 			glog.Errorf("proxy serve grpc with error: %v", err)
 		}
 
+		if context.String("watch-vm-console") != "" {
+			glog.Infof("watchConsole() sock: %s", context.String("watch-vm-console"))
+			err = watchConsole(context.String("watch-vm-console"))
+			if err != nil {
+				glog.Errorf("watchConsole() failed, err: %#v", err)
+				return err
+			}
+		}
+
```

Comment:
I don't understand "Without proxy serve grpc, the watchConsole will be blocked
at utils.UnixSocketConnect(console)." 
What's the dependence behind it?
And the code is moved after "s.Serve(l)", which means the proxy is done. We don't want to watch the console after the proxy is done.

The reason for the blocking at utils.UnixSocketConnect(console) might be the proxy didn't wait the READY message. If so, it is fixed by the first commit in this PR.

Reply:
@laijs I have checked code again. The correct change would be removing the " watchConsole(context.String("watch-vm-console"))". Because in lkvm driver the console socket
would not be created any more. Current watch vm console has used a channel to log VM message. This UnixSocketConnect will be blocked here forever. Move the watchConsole behind the "s.Serve(l)" just make watchConsole do not block the grpcSock creating. It just covers up the real issue. I will send another patch to remove the watchConsole from createProxy.

>I1117 10:54:54.422648   22017 json.go:194] SB[vm-ALZDANbZzu] hyperstart API version:4244, VM hyperstart API version: 4244
E1117 10:55:03.704558   22017 proxy.go:74] watchConsole() failed, err: &net.OpError{Op:"dial", Net:"unix", Source:net.Addr(nil), Addr:(*net.UnixAddr)(0x442024f1e0), Err:(*os.SyscallError)(0x442024f200)}

The function "newHyperstart" will wait grpcSock for 10 seconds.
		for i := 0; i < 500; i++ {
			if _, err := os.Stat(grpcSock); !os.IsNotExist(err) {
				break
			}
			time.Sleep(20 * time.Millisecond)
		}
But, from above log, you can see that, the watchConsole block the grpcSock create for 10 seconds.

Reply:
@laijs 
Maybe add some check code into watchConsole function is better than removing it.
Because maybe some hypervisors still use unix sock to watch VM console message:
>	_, err := os.Stat(console)
	if err != nil {
		if os.IsNotExist(err) {
			glog.Infof("Console sock %s isn't existed, using channel", console)
			return nil
		}
		return fmt.Errorf("%s existed, someone may be in service", console)
	}

Reply:
I think:
First, try to find the possible way to connect to the console for lkvm.
If there is none, the way could be not to pass the --watch-vm-console argument
to proxy when lkvm driver.

Reply:
That's make sense. I would do it. Thanks!